clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -1969,6 +1969,43 @@ def VecCreateOp : CIR_Op<"vec.create", [Pure]> {
   let hasVerifier = 1;
 }
 
+//===----------------------------------------------------------------------===//
+// VecInsertOp
+//===----------------------------------------------------------------------===//
+
+def VecInsertOp : CIR_Op<"vec.insert", [Pure,
+  TypesMatchWith<"argument type matches vector element type", "vec", "value",
+                 "cast<VectorType>($_self).getElementType()">,
+  AllTypesMatch<["result", "vec"]>]> {
+
+  let summary = "Insert one element into a vector object";
+  let description = [{
+    The `cir.vec.insert` operation replaces the element of the given vector at
+    the given index with the given value.  The new vector with the inserted
+    element is returned.
```

Comment:
Does the original vector become invalid, or is it preserved in its original state, or if referenced will it contain the inserted value?

Reply:
As far as I understood, if we didn't emit StoreOp, the old vector will be in its original state, but I will double check in IR and LLVM dialect

Reply:
The description in the incubator is confusing: we don't implicitly store. Perhaps rewrite to something like:
```
The `cir.vec.insert` operation produces a new vector by replacing the element of the input vector at `index` with `value`.
```